server/src/server.rs

```diff
@@ -805,18 +805,20 @@ where
 	<B as HttpBody>::Data: Send,
 {
 	let conn = hyper::server::conn::Http::new().serve_connection(socket, service).with_upgrades();
+	let stopped = stop_handle.shutdown();
 
-	tokio::pin!(conn);
+	tokio::pin!(stopped);
 
-	tokio::select! {
-		res = &mut conn => {
-			if let Err(e) = res {
-				tracing::warn!("HTTP serve connection failed {:?}", e);
-			}
-		}
-		_ = stop_handle.shutdown() => {
-			conn.graceful_shutdown();
+	let res = match future::select(conn, stopped).await {
+		Either::Left((conn, _)) => conn,
+		Either::Right((_, mut conn)) => {
+			Pin::new(&mut conn).graceful_shutdown();
+			conn.await
```

Comment:
this is the fix, we need to await for the conn future anyway not just call `graceful_shutdown` my bad

Reply:
See https://github.com/hyperium/hyper/issues/2939#issuecomment-1213481266

Reply:
Ah interesting! (I found https://docs.rs/hyper/latest/hyper/server/conn/struct.Connection.html#method.graceful_shutdown more useful)

Might be worth a comment next to these lines to note that since it's a slightly odd API?